### PR TITLE
Added Junit Tests

### PR DESCRIPTION
**Pull request by sonja-ct**
Added Junit tests for AboutEntry class
Added Junit tests for AboutScanner class
Added Junit tests for EventStoreQueryImpl class
Added Junit tests for EventStoreRecordCreatorImpl class
Added Junit Tests for EntityCache class
Added Junit Tests for KapuaCacheManager class
Added Junit Tests for NamedEntityCache class

Signed-off-by: Sonja <sonja.matic@comtrade.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
Hey @Coduz,
I have checked this PR and it looks Ok. Can you go through and merge it when you have time? Thanks!

File: commons/src/test/java/org/eclipse/kapua/commons/about/AboutEntryTest.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    @Test
+    public void aboutEntryLicenseTest() {
+        for (String name : names) {
+            for (String text : texts) {
+                for (URL url : urls) {
+                    AboutEntry.License license = new AboutEntry.License(name, text, url);
+                    assertEquals("Expected and actual values should be the same.", name, license.getName());
+                    assertEquals("Expected and actual values should be the same.", text, license.getText());
+                    assertEquals("Expected and actual values should be the same.", url, license.getUrl());
+                    assertNotNull("Null not expected.", license);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/about/AboutEntryTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/about/AboutEntryTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                    assertThat("AboutEntry.License object expected.", license, IsInstanceOf.instanceOf(AboutEntry.License.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/about/AboutEntryTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/about/AboutScannerTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.LinkedList;
+import java.util.List;
+
+@Category(JUnitTests.class)
+public class AboutScannerTest extends Assert {
+
+    @Test
+    public void scanTest() {
+        final AboutScanner scanner = AboutScanner.scan();
+        assertNotNull("Null not expected", scanner);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/about/AboutScannerTest.java:30]: For example in this case this makes sense.
There is no `new` and `AboutScanner.scan()` can return `null` even if is not designed to, if someone changes the implementation of `scan()`

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.math.BigInteger;
+
+@Category(JUnitTests.class)
+public class EventStoreQueryImplTest extends Assert {
+
+    @Test
+    public void eventStoreQueryImplTest1() {
+        EventStoreQueryImpl eventStoreQueryImpl = new EventStoreQueryImpl();
+        assertNotNull("Null not expected.", eventStoreQueryImpl);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        assertThat("EventStoreQueryImpl object expected.", eventStoreQueryImpl, IsInstanceOf.instanceOf(EventStoreQueryImpl.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        assertNotNull("Null not expected.", eventStoreQueryImpl.getSortCriteria());
+        assertThat("FieldSortCriteriaImpl object expected.", eventStoreQueryImpl.getSortCriteria(), IsInstanceOf.instanceOf(FieldSortCriteriaImpl.class));
+    }
+
+    @Test
+    public void eventStoreQueryImplTest2() {
+        KapuaId[] scopeIdList = {new KapuaIdStatic(BigInteger.ONE), new KapuaIdStatic(BigInteger.TEN), new KapuaIdStatic(BigInteger.ZERO)};
+
+        EventStoreQueryImpl eventStoreQueryImpl1 = new EventStoreQueryImpl(null);
+        assertNotNull("Null not expected.", eventStoreQueryImpl1);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        assertThat("EventStoreQueryImpl object expected.", eventStoreQueryImpl1, IsInstanceOf.instanceOf(EventStoreQueryImpl.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        assertNotNull("Null not expected.", eventStoreQueryImpl1.getSortCriteria());
+        assertThat("FieldSortCriteriaImpl object expected.", eventStoreQueryImpl1.getSortCriteria(), IsInstanceOf.instanceOf(FieldSortCriteriaImpl.class));
+        assertNull("Null expected.", eventStoreQueryImpl1.getScopeId());
+
+        for (KapuaId scopeId : scopeIdList) {
+            EventStoreQueryImpl eventStoreQueryImpl2 = new EventStoreQueryImpl(scopeId);
+            assertNotNull("Null not expected.", eventStoreQueryImpl2);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            assertThat("EventStoreQueryImpl object expected.", eventStoreQueryImpl2, IsInstanceOf.instanceOf(EventStoreQueryImpl.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreQueryImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.math.BigInteger;
+
+@Category(JUnitTests.class)
+public class EventStoreRecordCreatorImplTest extends Assert {
+    @Test
+    public void eventStoreRecordCreatorImplTest() {
+        KapuaId[] scopeIdList = {null, new KapuaIdStatic(BigInteger.ONE), new KapuaIdStatic(BigInteger.TEN), new KapuaIdStatic(BigInteger.ZERO)};
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java]: Same as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R103

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+        for (KapuaId scopeId : scopeIdList) {
+            EventStoreRecordCreatorImpl eventStoreRecordCreatorImpl = new EventStoreRecordCreatorImpl(scopeId);
+            assertNotNull("Null not expected.", eventStoreRecordCreatorImpl);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+            assertEquals("Expected and actual values should be the same.", scopeId, eventStoreRecordCreatorImpl.getScopeId());
+            assertThat("EventStoreRecordCreatorImpl object expected.", eventStoreRecordCreatorImpl, IsInstanceOf.instanceOf(EventStoreRecordCreatorImpl.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordCreatorImplTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/service/internal/cache/EntityCacheTest.java [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+        String idCacheName = "idCacheName";
+        Cache<Serializable, Serializable> expectedIdCache = KapuaCacheManager.getCache(idCacheName);
+        Cache<Serializable, Serializable> expectedListsCache = KapuaCacheManager.getCache(idCacheName + "_list");
+        Counter expectedCacheMiss = MetricServiceFactory.getInstance().getCounter("commons", "cache", "entity", "miss", "count");
+        Counter expectedCacheHit = MetricServiceFactory.getInstance().getCounter("commons", "cache", "entity", "hit", "count");
+        Counter expectedCacheRemoval = MetricServiceFactory.getInstance().getCounter("commons", "cache", "entity", "removal", "count");
+        NullPointerException nullPointerException = new NullPointerException();
+
+        EntityCache entityCache = new EntityCache(idCacheName);
+        assertNotNull(entityCache);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/service/internal/cache/EntityCacheTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

**Review by Coduz — CHANGES_REQUESTED** [review]

**Comment by sonja-ct** [comment]
Hey @Coduz,
I made all the requested changes, if you can take a look, I hope everything is ok now. 
Thanks, Sonja

**Review by Coduz — APPROVED** [review]

**Pull request merged**